lib/puppet/provider/iis_application_pool/webadministration.rb

```diff
@@ -24,6 +24,7 @@ def create
     result   = self.class.run(inst_cmd)
     Puppet.err "Error creating apppool: #{result[:errormessage]}" unless result[:exitcode] == 0
     Puppet.err "Error creating apppool: #{result[:errormessage]}" unless result[:errormessage].nil?
+    @resource.provider = self.class.instances.find{|prov| prov.name == @resource[:name]}
```

Comment:
That was my comment when screen sharing; that this will run once per create and that could be slow, but I interpreted the answer as "don't worry about that." I have no idea how to make it faster... does "filtering" make it run faster?